subsys/net/ip/tcp2_priv.h

```diff
@@ -22,10 +22,14 @@
 
 #define tcp_slist(_slist, _op, _type, _link)				\
 ({									\
+	k_mutex_lock(&tcp_lock, K_FOREVER);				\
+									\
 	sys_snode_t *_node = sys_slist_##_op(_slist);			\
 									\
 	_type * _x = _node ? CONTAINER_OF(_node, _type, _link) : NULL;	\
 									\
+	k_mutex_unlock(&tcp_lock);					\
+									\
```

Comment:
`tcp_slist()` is used mainly on `&conn->send_queue`. So wy do we need to take a global lock instead of `conn->lock` (for example)?

Reply:
Yep, conn->lock is enough here.